Carmen Cortez_Assignment 4_PCA.Rmd

```diff
@@ -0,0 +1,154 @@
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+```
+## Now print out the eigenvectors (often called loadings) for the components you generated:
+
+```{r}
+pca$rotation
+#Examine the eigenvectors, notice that they are a little difficult to interpret. It is much easier to make sense of them if we make them proportional within each component
+
+loadings <- abs(pca$rotation) 
+#abs() will make all eigenvectors positive
+
+sweep(loadings, 2, colSums(loadings), "/") 
+#sweep() computes each row as a proportion of the column. (There must be a way to do this with dplyr()?)
+
+#Now examine your components and try to come up with substantive descriptions of what some might represent? The correlations between the principal components and the original variables are copied into the table below. The first principal componet is strongly correctlated with two of the original values. The first component increases with increasing mean_hint and mean_attempt scores. It would follow that students with higher number of hints received would perform better. The second principal component increases with only one of the values, increasing prior_percent_correct. This component can be viewed as a measure of how well student perform when the percentage of problems a student has answered correctly prior to this session is higher. The third component decreases with decreasing mean_confidence. This suggests that performance decreases when average confidence each student has in their ability to answer the problems decreases. The fourth principal component increases only with only one of the values, decreasing prior_prob_count. This measure can be viewed as how poorly students fair when the number of problems students have answered in the system prior to this session decreases. The fifth principal component is strongly correlated with two of the original values, problems attempted and means_attempt. It would make sense to see an increase in performance when the average number of correct answers on first attempt increases and the average number of attempts decreases. In contrast the fourth principal, the sixth principal increases with only one of the values, increasing mean_hint. This indicates that students tend to perform better when they have access to additional support.
+
+#You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
+
+biplot(pca)
+#Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
+
```

Comment:
For this part we want to create a composite variable that represents the different components we think are relevant:

This will get the proportions:
D5 <- mutate_all(as.data.frame(loadings), funs((.)/sum(.)))

Then we can do some weighting based on those (this is an example of how to do it manually):
D2 <- as.data.frame(D2)
D2$aggressiveness <- D2$prior_prob_count*0.12423113 + D2$prior_percent_correct*0.08035956 + D2$problems_attempted*0.21744737 + D2$mean_hint*0.30223780 + D2$mean_attempt*0.25863458 + D2$mean_confidence*0.01708956